File: Aplikacija/ElfakInfostud/client/src/app/domain/post.ts
export interface Post {
    id:string;
    title: string;
    content: string;
    category: string;
    date: Date;
    numberOfViews: number;
    authorUsername: string;
    approved: boolean;
    authorImage: string;
    likes: string[];
}

export class Post implements Post {
    constructor(post : PostForm ){
        Object.assign(this,post);
    }
    


}

export class PostForm {
    id?:string = undefined;
    title:string = '';
    content:string = '';
    category: string = '';
    numberOfViews:number = 0;
    likes: string[] = [];

    constructor(post? : Post){
        if(post){
            this.id = post.id;
            this.title = post.title;
            this.content = post.content;
            this.category = post.category;
           this.numberOfViews = post.numberOfViews;
        }
    }
}

File: Aplikacija/ElfakInfostud/Domain/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public string Category { get; set; }
        public DateTime Date { get; set; } = DateTime.Now;
        public int NumberOfViews { get; set; }
        public bool Approved { get; set; }
        
        public User Author { get; set; }

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Like> Likes { get; set; } = new List<Like>();

    }
}